common/src/main/java/org/apache/falcon/metadata/InstanceRelationshipGraphBuilder.java

```diff
@@ -250,20 +250,15 @@ public void addEvictedInstance(WorkflowExecutionContext context) throws FalconEx
                     feedName, evictedFeedInstancePath, clusterName);
             String feedInstanceName = getFeedInstanceName(feedName, clusterName,
                     evictedFeedInstancePath, context.getNominalTimeAsISO8601());
-            Vertex feedInstanceVertex = findVertex(feedInstanceName,
-                    RelationshipType.FEED_INSTANCE);
-
-            LOG.info("Vertex exists? name={}, type={}, v={}",
-                    feedInstanceName, RelationshipType.FEED_INSTANCE, feedInstanceVertex);
-            if (feedInstanceVertex == null) { // No record of instances NOT generated by Falcon
-                LOG.info("{} instance vertex {} does not exist, add it",
-                        RelationshipType.FEED_INSTANCE, feedInstanceName);
-                feedInstanceVertex = addFeedInstance(// add a new instance
-                        feedInstanceName, context, feedName, clusterName);
-            }
+            Vertex feedInstanceVertex = addFeedInstance(
+                    feedInstanceName, context, feedName, clusterName);
 
+            Map<RelationshipProperty, String> properties = new HashMap<RelationshipProperty, String>();
```

Comment:
Created a new function edgePropertiesForIndexing to generate "properties". Note that the property TIMESTAMP is inherited from the original code, but not for indexing, so it is added after calling edgePropertiesForIndexing. Also, edgePropertiesForIndexing is used in other places that doesn't need TIMESTAMP (e.g. addProcessInstance).